### PR TITLE
Unitest rewrite to pytest

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: pulpcore/tests/unit/test_settings.py [hunk 1]
```diff
@@ -1,59 +1,54 @@
...
-        """Test that API_ROOT validation checks for beginning and ending '/'."""
-        msettings = settings.from_env("development", keep=True, validators=settings.validators)
-        msettings.set("API_ROOT", "/hi/there")
-        with self.assertRaises(ValidationError):
-            msettings.validators.validate()
-
-        msettings.set("API_ROOT", "hi/there/")
-        with self.assertRaises(ValidationError):
-            msettings.validators.validate()
+def test_content_origin(settings):
```
Comment (lubosmj) [inline pulpcore/tests/unit/test_settings.py:5]: Can we benefit from the `settings` fixture in functional tests as well?

I can remember that two of us were having problems with `pulp_settings` during its initialization in oci-env:
```
======================================================= ERRORS =======================================================
___________________________________ ERROR at setup of test_rbac_repository_content ___________________________________

    @pytest.fixture(scope="session")
    def pulp_settings():
        import django
    
>       django.setup()

src/pulpcore/pulpcore/tests/functional/__init__.py:827: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
usr/local/lib/python3.8/site-packages/django/__init__.py:24: in setup
    apps.populate(settings.INSTALLED_APPS)
usr/local/lib/python3.8/site-packages/django/apps/registry.py:114: in populate
    app_config.import_models()
usr/local/lib/python3.8/site-packages/django/apps/config.py:301: in import_models
    self.models_module = import_module(models_module_name)
usr/lib64/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
src/pulp_container/pulp_container/app/models.py:31: in <module>
    from . import downloaders
src/pulp_container/pulp_container/app/downloaders.py:16: in <module>
    class RegistryAuthHttpDownloader(HttpDownloader):
src/pulp_container/pulp_container/app/downloaders.py:24: in RegistryAuthHttpDownloader
    token_lock = asyncio.Lock()
usr/lib64/python3.8/asyncio/locks.py:164: in __init__
    self._loop = events.get_event_loop()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <asyncio.unix_events._UnixDefaultEventLoopPolicy object at 0x7f19fdd70f10>

    def get_event_loop(self):
        """Get the event loop for the current context.
    
        Returns an instance of EventLoop or raises an exception.
        """
        if (self._local._loop is None and
                not self._local._set_called and
                isinstance(threading.current_thread(), threading._MainThread)):
            self.set_event_loop(self.new_event_loop())
    
        if self._local._loop is None:
>           raise RuntimeError('There is no current event loop in thread %r.'
                               % threading.current_thread().name)
E           RuntimeError: There is no current event loop in thread 'MainThread'.
```

File: pulpcore/tests/unit/viewsets/test_viewset_base.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    """
+    Test that get_serializer_class() raises an AssertionError if you don't define the
+    serializer_class attribute.
+    """
+
+    class TestTaskViewSet(viewsets.NamedModelViewSet):
+        minimal_serializer_class = serializers.MinimalTaskSerializer
+
+    with pytest.raises(AssertionError):
+        TestTaskViewSet().get_serializer_class()
```
Comment (lubosmj) [inline pulpcore/tests/unit/viewsets/test_viewset_base.py:54]: If this tests Django's facility, we can remove it.

**Review by lubosmj — APPROVED** [review]
Great job!

I have added a couple of comments. Feel free to ignore them. I think we can merge the PR as it is.

File: pulpcore/tests/unit/roles/test_roles.py [hunk 3]
```diff
@@ -13,84 +12,115 @@
...
+    remove_role("role1", user, repository)
+
+
+def test_user_role(user, repository, role1):
+    assign_role("role1", user)
+    assert user.has_perm("core.view_repository")
+    assert not user.has_perm("core.view_repository", repository)
+    assert user.get_all_permissions() == {"core.view_repository"}
+    assert user.get_all_permissions(repository) == set()
+    remove_role("role1", user)
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/tests/unit/roles/test_roles.py:89]: It does not really matter much. The whole test runs in a non commited transaction. So this is not for cleanup, but for testing that remove_role does not fail.

File: pulpcore/tests/unit/roles/test_roles.py [hunk 4]
```diff
@@ -13,84 +12,115 @@
...
-            ),
-            {self.remote.pk, self.remote2.pk},
+    ) == {repository.pk}
+    assert set(
+        get_objects_for_user(user, "core.view_remote", Remote.objects.all()).values_list(
+            "pk", flat=True
         )
-        remove_role("role2", self.group)
+    ) == {remote.pk, remote2.pk}
+    remove_role("role2", group)
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/tests/unit/roles/test_roles.py:126]: Let's add it. Again this is not for cleanup...

File: pulpcore/tests/unit/test_settings.py (hunk 1 above)
Comment (mdellweg) [reply pulpcore/tests/unit/test_settings.py:5]: I'm not sure.

File: pulpcore/tests/unit/viewsets/test_viewset_base.py (hunk 2 above)
Comment (mdellweg) [reply pulpcore/tests/unit/viewsets/test_viewset_base.py:54]: `minimal_serializer_class` is a pulpcore feature. ;)

Comment (lubosmj) [reply pulpcore/tests/unit/viewsets/test_viewset_base.py:54]: But, `serializer_class` is not.

Comment (mdellweg) [reply pulpcore/tests/unit/viewsets/test_viewset_base.py:54]: I assume the test is targeted a this line: https://github.com/pulp/pulpcore/blob/main/pulpcore/app/viewsets/base.py#L103

Comment (lubosmj) [reply pulpcore/tests/unit/viewsets/test_viewset_base.py:54]: Hmm, you are right, again. 🥱

File: pulpcore/tests/unit/test_cache.py [hunk 5]
```diff
@@ -1,102 +1,108 @@
...
+    assert ret is None
+
+
+def test_basic_expires(pulp_redisdb):
+    """Tests setting values with expiration times"""
+    cache = Cache()
+    cache.set("key", "hi", expires=4)
+    ret = cache.get("key")
+    assert ret == b"hi"
+    sleep(5)
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/tests/unit/test_cache.py:52]: I did not design this test. Let's refer to @gerrod3 .

[the remark this answers is not in the document]
Comment (gerrod3) [reply pulpcore/tests/unit/test_cache.py:52]: If you change the `expires` time to something shorter than yes. Else we have have sleep as long as the key is not expired.

**Review by gerrod3 — APPROVED** [review]

**Pull request merged**